### PR TITLE
Add some documentation/summaries to meteor type defs

### PR DESCRIPTION
**Pull request by ToastHawaii**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://docs.meteor.com/#/full/, https://github.com/meteor/meteor/
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/meteor/tsconfig.json [hunk 1]
```diff
@@ -27,4 +27,4 @@
         "test/react-meteor-data-tests.tsx",
         "test/meteor-mongo-tests.ts"
     ]
-}
+}
```
Comment (radekmie) [inline types/meteor/tsconfig.json:30]: I'd revert this change for two reasons:
1. We'll need to go through a much longer review to merge this PR.
2. All other files have an additional new line at the end.

File: types/meteor/check.d.ts [hunk 2]
```diff
@@ -1,45 +1,84 @@
...
+            ? Function
+            : T extends undefined | null | string | number | boolean
+            ? T
+            : T extends new (...args: any[]) => infer U
+            ? U
+            : T extends [Pattern]
+            ? PatternMatch<T[0]>[]
+            : T extends { [key: string]: Pattern }
+            ? { [K in keyof T]: PatternMatch<T[K]> }
+            : unknown;
```
Comment (radekmie) [inline types/meteor/check.d.ts:44]: In my subjective opinion, the previous version was much better. I know it doesn't follow any strict formatting rules (I guess you've just used ESLint, Prettier, or both), but in this very case, I'd keep these in a one-line-per-branch manner.

File: types/meteor/accounts-base.d.ts [hunk 3]
```diff
@@ -54,15 +57,29 @@ declare module "meteor/accounts-base" {
...
+        function changePassword(
+            oldPassword: string,
+            newPassword: string,
+            callback?: (error?: Error | Meteor.Error | Meteor.TypedError) => void,
+        ): void;
+
+        function forgotPassword(
+            options: {
+                email?: string;
+            },
```
Comment (radekmie) [inline types/meteor/accounts-base.d.ts:69]: In a couple of places, the formatting expands small and simple objects like this one into multiple ones, just because it was a multi-line object before. How about wrapping them back into one line? It'd shave off a couple of lines from the files and make them _slightly_ more comprehensible.

File: types/meteor/meteor.d.ts [hunk 4]
```diff
@@ -64,15 +106,32 @@ declare module "meteor/meteor" {
...
             isSimulation: boolean;
+            /** The id of the user that made this method call, or `null` if no user was logged in. */
             userId: string | null;
+            /** Access inside a method invocation. The connection that this method was received on. `null` if the method is not associated with a connection, eg. a server initiated method call. Calls to methods made from a server method which was in turn initiated from the client share the same `connection`. */
             connection: Connection | null;
+            /**
+             * Set the logged in user.
+             * @param userId The value that should be returned by `userId` on this connection.
+             */
             setUserId(userId: string): void;
```
Comment (radekmie) [inline types/meteor/meteor.d.ts:119]: You've corrected the `ddp.d.ts` file to allow `null`s. I guess applies here as well?

File: types/meteor/mongo.d.ts [hunk 5]
```diff
@@ -1,200 +1,338 @@
...
+            | 'timestamp'
+            | 18
+            | 'long'
+            | 19
+            | 'decimal'
+            | -1
+            | 'minKey'
+            | 127
+            | 'maxKey'
+            | 'number';
```
Comment (radekmie) [inline types/meteor/mongo.d.ts:52]: Same as in `check.d.ts`, I'd keep the old formatting here. It clearly presented that the string values were aliases of the numerical ones.

**Review by radekmie — CHANGES_REQUESTED** [review]

**Comment by ToastHawaii** [comment]
@radekmie Thank you for your review, I hope I have addressed all your concerns.

About the formatting: I have used prettier as described here https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes

I added some `// prettier-ignore` to prevent some lines from being formatted.

File: types/meteor/globals/mongo.d.ts [hunk 6]
```diff
@@ -239,3 +390,14 @@ declare module Mongo {
         transform?: Function | null;
     }
 }
+
+declare interface MongoConnection {
+    db: any;
+    client: any;
+}
```
Comment (radekmie) [inline types/meteor/globals/mongo.d.ts:397]: Why `any`? It works correctly in `types/meteor/mongo.d.ts` with proper types:
```ts
    interface MongoConnection {
        db: MongoDb;
        client: MongoClient;
    }
```

**Review by radekmie — COMMENTED** [review]
Prettier will never wrap an object back into one line. For example, this code will be preserved as it is:
```ts
type A = { a: string };
type B = {
  b: string;
};
```

And as there are quite a few objects in these typings, it'd get rid of a lot of lines.

File: types/meteor/globals/mongo.d.ts (hunk 6 above)
Comment (ToastHawaii) [reply types/meteor/globals/mongo.d.ts:379]: I tried it with `/// <reference types="mongodb" />`
but it didn't work so I described it as `any;` to get further.

**Comment by ToastHawaii** [comment]
> Prettier will never wrap an object back into one line. For example, this code will be preserved as it is:
> 
> ```ts
> type A = { a: string };
> type B = {
>   b: string;
> };
> ```
> 
> And as there are quite a few objects in these typings, it'd get rid of a lot of lines.

@radekmie Thank you for your explanation. Now I understand how prettier wrapps.

I have reduced types with only one property to one line if they do not have a long description.

**Review by radekmie — APPROVED** [review]

**Comment by ToastHawaii** [comment]
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/54414
